File: prng/ruby/prng.rb
#!/usr/bin/env ruby

P = 295075153
Captured = [
    210205973,
    22795300,
    58776750,
    121262470,
    264731963,
    140842553,
    242590528,
    195244728,
    86752752
]

class Prng
    def initialize(x, y)
      @x = x
      @y = y
    end

    def nextPrng()
        @x = (2 * @x + 5) % P
        @y = (3 * @y + 7) % P
        return @x ^ @y
    end

    def verify()
        q = Prng.new(@x, @y)
        Captured[2..-1].each do |c|
            return false if q.nextPrng() != c
        end
        puts q.nextPrng()
	return true
    end
end


t = Time.now
0.upto(P) do |x|
    p = Prng.new(x, Captured[0] ^ x)
    break if p.nextPrng() == Captured[1] && p.verify()
end
puts "#{Time.now - t} seconds"

File: prng/python/prng.py
#!/usr/bin/env python3

import time

P = 295075153
captured = [
    210205973,
    22795300,
    58776750,
    121262470,
    264731963,
    140842553,
    242590528,
    195244728,
    86752752
]

class Prng(object):
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def nextPrng(self):
        self.x = (2 * self.x + 5) % P
        self.y = (3 * self.y + 7) % P
        return self.x ^ self.y

    def verify(self):
        q = Prng(self.x, self.y)
        for c in captured[2:]:
            if c != q.nextPrng():
                return False
        print(q.nextPrng())
        return True

t = time.clock()
for x in range(P):
    p = Prng(x, captured[0] ^ x)
    if p.nextPrng() == captured[1] and p.verify():
        break;

print(time.clock() - t, "seconds")
